README.md

```diff
@@ -120,160 +120,165 @@ public class MoveSystem : ISystem
 }
 ```
 
-## Query
-
-```csharp
-// every entity that has a Name and Age component and owes bob money.
-var appleLovers = commands.Query()
-    .Has<Name, Age>()
-    .Has<Owes>(bob);
-```
 
-Running a System
+### Running a System
 
 ```csharp
-// create an instance of our system
+// Create an instance of your system.
 var moveSystem = new MoveSystem();
 
-// run the system
-// the system will apply to all entities of that world
+// Run the system.
+// The system will match all entities of the world you enter as the parameter.
 moveSystem.Run(world);
 
-// we can run it however often you want
-// usually systems are run in a game loop
+// You can run a system as many times as you like.
 moveSystem.Run(world);
 moveSystem.Run(world);
 moveSystem.Run(world);
+
+// Usually, systems are run once a frame, inside your game loop.
+```
+
+## Query
+
+```csharp
+// You can create complex, expressive queries.
+// Here, we request every entity that has a Name and Age component and owes money to Bob.
+var appleLovers = commands.Query()
+    .Has<Name, Age>()
+    .Has<Owes>(bob);
 ```
 
 ## Triggers
 
 ```csharp
-// triggers are again just structs. They are basically components internally
+// Triggers are also just structs and very similar to components.
+// They act much like a simplified, ECS version of C# events.
 struct MyTrigger { }
 
-// send a bunch of triggers
+// You can send a bunch of triggers inside of a system.
 commands.Send<MyTrigger>();
 commands.Send<MyTrigger>();
 commands.Send<MyTrigger>();
 
-
-// in any system, you can now receive triggers
+// In any system, including the origin system, you can now receive these triggers.
 commands.Receive((MyTrigger e) =>
 {
-    Console.WriteLine("A Trigger!");
+    Console.WriteLine("It's a trigger!");
 });
-// note that triggers only live for 2 frames and can only be received ONCE per SYSTEM
 
 // Output:
 // A Trigger!
 // A Trigger!
 // A Trigger!
+
+// NOTE: Triggers live until the end of the next frame, to make sure every system receives them.
+// Each trigger is always received exactly ONCE per system.
 ```
 
-## Build-in Triggers
+## Built-in Triggers
 
 ```csharp
 var entity = commands.Spawn();
 
-// normally you add components like this. No triggers are spawned by default
-entity.Add<Name>(new Name("Walter"));
+// When you add a component, usually no triggers are created.
+// You can pass in an optional parameter 'spawnTrigger' to spawn an Added<T> trigger.
+entity.Add(new Name("Walter")); // No trigger.
+entity.Add(new Name("Walter"), true); // Trigger.
 
-// you can pass in an optional parameter 'spawnTrigger' 
-// to spawn an Added<T> trigger
-entity.Add<Old>(true);
+// This also works for "tag" components without any data.
+entity.Add<Old>(); // No trigger.
+entity.Add<Old>(true); // Trigger.
```

Comment:
this also works with components that have data, but where no initial values are needed.
```csharp
public struct Purse { public int Value; }

entity.Add<Purse>(); // initialized with Value = 0
```

Reply:
> this also works with components that have data, but where no initial values are needed.
> 
> ```cs
> public struct Purse { public int Value; }
> 
> entity.Add<Purse>(); // initialized with Value = 0
> ```

So it works both for:
- "Tag" components without any data
- Components that have data, but where no initial values are needed

Is this correct?

Reply:
Yes, and it also works for components where initial values are needed.